### PR TITLE
Init commit. 

### PR DESCRIPTION
**Pull request by jherrmann06**
Code is not ready yet, but it is as complete as it can be without a swerve chassis ready.

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    public static final double DRIVETRAIN_TRACKWIDTH_METERS = 0.599; //Measured in cad; FIXME Measure and set trackwidth
+    /**
+     * The front-to-back distance between the drivetrain wheels.
+     *
+     * Should be measured from center to center.
+     */
+    public static final double DRIVETRAIN_WHEELBASE_METERS = 0.601; //Mesasured in cad; FIXME Measure and set wheelbase
+
+    public static final int DRIVETRAIN_PIGEON_ID = 0; // FIXME Set Pigeon ID
```
Comment (jdlanning) [inline src/main/java/frc/robot/Constants.java]: We can delete the constants we arnt using

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ * This class is where the bulk of the robot should be declared. Since Command-based is a
+ * "declarative" paradigm, very little robot logic should actually be handled in the {@link Robot}
+ * periodic methods (other than the scheduler calls). Instead, the structure of the robot (including
+ * subsystems, commands, and button mappings) should be declared here.
+ */
+public class RobotContainer {
+  // The robot's subsystems and commands are defined here...
+  private final DrivetrainSubsystem m_drivetrainSubsystem = new DrivetrainSubsystem();
+
+  private final XboxController m_controller = new XboxController(0);
```
Comment (jdlanning) [inline src/main/java/frc/robot/RobotContainer.java]: Lets add a constant for 0

File: src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+          new Translation2d(-DRIVETRAIN_TRACKWIDTH_METERS / 2.0, DRIVETRAIN_WHEELBASE_METERS / 2.0),
+          // Back right
+          new Translation2d(-DRIVETRAIN_TRACKWIDTH_METERS / 2.0, -DRIVETRAIN_WHEELBASE_METERS / 2.0)
+  );
+
+  // By default we use a Pigeon for our gyroscope. But if you use another gyroscope, like a NavX, you can change this.
+  // The important thing about how you configure your gyroscope is that rotating the robot counter-clockwise should
+  // cause the angle reading to increase until it wraps back over to zero.
+  // FIXME Remove if you are using a Pigeon
+  private final PigeonIMU m_pigeon = new PigeonIMU(DRIVETRAIN_PIGEON_ID);
```
Comment (jdlanning) [inline src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java]: We can remove the Pigeon

File: src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    //   Your module has two Falcon 500s on it. One for steering and one for driving.
+    //
+    // Mk3SwerveModuleHelper.createNeo(...)
+    //   Your module has two NEOs on it. One for steering and one for driving.
+    //
+    // Mk3SwerveModuleHelper.createFalcon500Neo(...)
+    //   Your module has a Falcon 500 and a NEO on it. The Falcon 500 is for driving and the NEO is for steering.
+    //
+    // Mk3SwerveModuleHelper.createNeoFalcon500(...)
+    //   Your module has a NEO and a Falcon 500 on it. The NEO is for driving and the Falcon 500 is for steering.
```
Comment (jdlanning) [inline src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java]: We can do a little clean up here as well for the commented code while we wait on the chassis to be ready.

File: src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+                    .withSize(2, 4)
+                    .withPosition(4, 0),
+            Mk3SwerveModuleHelper.GearRatio.STANDARD,
+            BACK_LEFT_MODULE_DRIVE_MOTOR,
+            BACK_LEFT_MODULE_STEER_MOTOR,
+            BACK_LEFT_MODULE_STEER_ENCODER,
+            BACK_LEFT_MODULE_STEER_OFFSET
+    );
+
+    m_backRightModule = Mk3SwerveModuleHelper.createNeo(
```
Comment (jdlanning) [inline src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java]: I believe we have the Mk4 modules. @smitty1707 can confirm.

**Comment by ASmith1675** [comment]
Yes. Mk4 modules. It's possible that the code is the same as for mk3 I
think there were just mechanical improvements.

File: src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java [hunk 6]
```diff
@@ -1,18 +1,35 @@
...
+import static frc.robot.Constants.DRIVETRAIN_TRACKWIDTH_METERS;
+import static frc.robot.Constants.DRIVETRAIN_WHEELBASE_METERS;
+import static frc.robot.Constants.FRONT_LEFT_MODULE_DRIVE_MOTOR;
+import static frc.robot.Constants.FRONT_LEFT_MODULE_STEER_ENCODER;
+import static frc.robot.Constants.FRONT_LEFT_MODULE_STEER_MOTOR;
+import static frc.robot.Constants.FRONT_LEFT_MODULE_STEER_OFFSET;
+import static frc.robot.Constants.FRONT_RIGHT_MODULE_DRIVE_MOTOR;
+import static frc.robot.Constants.FRONT_RIGHT_MODULE_STEER_ENCODER;
+import static frc.robot.Constants.FRONT_RIGHT_MODULE_STEER_MOTOR;
+import static frc.robot.Constants.FRONT_RIGHT_MODULE_STEER_OFFSET;
```
Comment (jdlanning) [inline src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java:20]: Overall looks good. Only thing that caught my eye was these imports looks a little verbose (separate import for each constant seems overkill). It should be possible to just import all constants in one line

**Review by jdlanning — APPROVED** [review]
My one comment is very minor so just keep it in mind if we end up doing more cleanup . I will merge so we can start making smaller PRs

**Pull request merged**